scripts/cmake/common.cmake

```diff
@@ -72,6 +72,50 @@ function(doctest_add_test)
     doctest_add_test_impl(${ARGN} JUNIT_OUTPUT)
 endfunction()
 
+# Unlike doctest_add_test we need to add doctest as a regular test
+# I mean test which output is not checked, but instead we check retcode
```

Comment:
Why not just check for no output?

Reply:
Because NO_OUTPUT mode is either broken or intended for something else.
See https://github.com/doctest/doctest/pull/590#issue-1105153070
I tried to use it, and as a result, I see "Status: FAILURE!" in the console, but CTest entry is completed just fine, no error is emitted. Because it seems that only two types of errors are allowed now:
1. Compilation of test is failed (possibly due to warnings)
2. Executable output differs from approved one (it seems that NO_OUTPUT disables exactly this)
The case when we have doctest based executable and if some of the tests fail, then CTest fails is not an option currently.

Reply:
Actually, I can accept the game rules and make approval test. For which only 
```
[doctest] doctest version is "2.4.8"
[doctest] run with "--help" for options
===============================================================================
[doctest] test cases: 0 | 0 passed | 0 failed | 12 skipped
[doctest] assertions: 0 | 0 passed | 0 failed |
[doctest] Status: SUCCESS!
```
output is allowed. But I don't like such approach.

Reply:
Imo that approach is preferable since I think cmake should remain minimal.

Reply:
Ok. I'll try to get it work.

Reply:
Done. But at what cost? 😭